pkg/types/spec/validations.go

```diff
@@ -817,28 +813,53 @@ func validatePredictor(
 func validateMultiModelsFields(api *userconfig.API) error {
 	predictor := api.Predictor
 
-	if len(predictor.Models.Paths) == 0 && predictor.Models.Dir == nil {
-		return errors.Wrap(ErrorSpecifyOneOrTheOther(userconfig.ModelsPathsKey, userconfig.ModelsDirKey), userconfig.ModelsKey)
+	var models *userconfig.MultiModels
+	if api.Predictor.Models != nil {
+		if api.Predictor.Type == userconfig.PythonPredictorType {
+			return ErrorFieldNotSupportedByPredictorType(userconfig.ModelsKey, api.Predictor.Type)
+		}
+		models = api.Predictor.Models
+	}
+	if api.Predictor.MultiModelReloading != nil {
+		if api.Predictor.Type != userconfig.PythonPredictorType {
+			return ErrorFieldNotSupportedByPredictorType(userconfig.MultiModelReloadingKey, api.Predictor.Type)
+		}
+		models = api.Predictor.MultiModelReloading
+	}
+	if models == nil && api.Predictor.Type != userconfig.PythonPredictorType {
+		return ErrorFieldMustBeDefinedForPredictorType(userconfig.ModelsKey, api.Predictor.Type)
+	}
+
+	if models.Path == nil && len(models.Paths) == 0 && models.Dir == nil {
+		return errors.Wrap(ErrorSpecifyOnlyOneField(userconfig.ModelsPathKey, userconfig.ModelsPathsKey, userconfig.ModelsDirKey), userconfig.ModelsKey)
+	}
+	if models.Path != nil && (len(models.Paths) > 0 || models.Dir != nil) {
```

Comment:
I think this should be three ifs, e.g.

`if models.Path != nil && len(models.Paths) > 0 {`

`if models.Dir != nil && len(models.Paths) > 0 {`

`if models.Dir != nil && models.Path != nil {`

Reply:
Consider adding a check if they are all specified

Reply:
Added the check if they are all specified as well.